src/compiler.ts

```diff
@@ -9941,7 +9941,13 @@ export class Compiler extends DiagnosticEmitter {
       case TypeKind.EQREF:
       case TypeKind.DATAREF:
       case TypeKind.I31REF: {
-        return module.ref_is(RefIsOp.RefIsNull, expr);
+        // Needs to be true when the ref is _not_ null.
+        // Therefore: 1 - <ref.is_null>
+        // Therefore: !(<ref.is_null>)
+        return type.size == 64
+          ? module.unary(UnaryOp.EqzI64, module.ref_is(RefIsOp.RefIsNull, expr))
+          : module.unary(UnaryOp.EqzI32, module.ref_is(RefIsOp.RefIsNull, expr));
```

Comment:
Afaik `ref.is_null` returns an `i32`.
```suggestion
        return module.unary(UnaryOp.EqzI32, module.ref_is(RefIsOp.RefIsNull, expr));
```

Reply:
ooh yeah you are right